diff --git a/sqlite3.go b/sqlite3.go
index 5ac9570..ae76365 100644
--- a/sqlite3.go
+++ b/sqlite3.go
@@ -1042,6 +1045,10 @@ func (d *SQLiteDriver) Open(dsn string) (driver.Conn, error) {
 	vfsName := ""
 	var cacheSize *int64
 
+	// SQLCipher PRAGMA's
+	pragmaKey := ""
+	pragmaCipherPageSize := -1
+
 	pos := strings.IndexRune(dsn, '?')
 	if pos >= 1 {
 		params, err := url.ParseQuery(dsn[pos+1:])
@@ -1380,6 +1387,20 @@ func (d *SQLiteDriver) Open(dsn string) (driver.Conn, error) {
 			vfsName = val
 		}
 
+		// _pragma_key
+		if val := params.Get("_pragma_key"); val != "" {
+			pragmaKey = val
+		}
+
+		// _pragma_cipher_page_size
+		if val := params.Get("_pragma_cipher_page_size"); val != "" {
+			pageSize, err := strconv.Atoi(val)
+			if err != nil {
+				return nil, fmt.Errorf("sqlite3: _pragma_cipher_page_size cannot be parsed: %s", err)
+			}
+			pragmaCipherPageSize = pageSize
+		}
+
 		if !strings.HasPrefix(dsn, "file:") {
 			dsn = dsn[:pos]
 		}
@@ -1425,6 +1446,25 @@ func (d *SQLiteDriver) Open(dsn string) (driver.Conn, error) {
 		return nil
 	}
 
+	// _pragma_key
+	if pragmaKey != "" {
+		query := fmt.Sprintf("PRAGMA key = \"%s\";", pragmaKey)
+		if err := exec(query); err != nil {
+			C.sqlite3_close_v2(db)
+			return nil, err
+		}
+	}
+
+	// _pragma_cipher_page_size
+	if pragmaCipherPageSize != -1 {
+		query := fmt.Sprintf("PRAGMA cipher_page_size = %d;",
+			pragmaCipherPageSize)
+		if err := exec(query); err != nil {
+			C.sqlite3_close_v2(db)
+			return nil, err
+		}
+	}
+
 	// USER AUTHENTICATION
 	//
 	// User Authentication is always performed even when
